File: PULSE.Services/Data/Orders/OrderHeader.cs
using PULSE.Model;

namespace PULSE.Services.Data
{
    public partial class OrderHeader : BaseEntity
    {
        public OrderHeader()
        {
            OrderDetails = new HashSet<OrderDetail>();
        }

        public string? OrderNumber { get; set; }
        public int? CustomerId { get; set; }
        public int? StaffId { get; set; }
        public int? PaymentId { get; set; }
        public int? ShippingInfoId { get; set; }
        public OrderState? Status { get; set; }
        public DateTime? TimeProcessed { get; set; }
        public DateTime? TimePacked { get; set; }
        public DateTime? TimeShipped { get; set; }
        public DateTime? TimeDelivered { get; set; }
        public bool? OnlineOrder { get; set; }
        public bool? Delivery { get; set; }
        public decimal? ShippingConst { get; set; }

        public virtual Customer? Customer { get; set; }
        public virtual staff? Staff { get; set; }
        public virtual Payment? Payment { get; set; }
        public virtual ShippingInfo? ShippingInfo { get; set; }
        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
        
        public decimal Total()
        {
            decimal result = 0;

            foreach (var item in OrderDetails)
            {
                result += (item.UnitPrice ?? 0) * (item.Quantity ?? 0);
            }

            return result;
        }
    }
}

File: pulse_admin/lib/api/types/orders/order_header.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pulse_admin/api/types/abstract/base_model.dart';
import 'package:pulse_admin/api/types/generic/shipping_info.dart';
import 'package:pulse_admin/api/types/orders/order_detail.dart';
import 'package:pulse_admin/api/types/user/customer.dart';
import 'package:pulse_admin/api/types/user/staff.dart';

part 'order_header.g.dart';

enum OrderStatus {
  initial,
  cart,
  draft,
  processed,
  packed,
  shipped,
  collected,
  delivered,
  cancelled
}

extension Name on OrderStatus {
  String getName() {
    return "${name.substring(0, 1).toUpperCase()}${name.substring(1)}";
  }

  String getNameUppercase() {
    return name.toUpperCase();
  }
}

@JsonSerializable()
class OrderHeader extends BaseModel {
  String? orderNumber;
  int? customerId;
  int? staffId;
  int? paymentId;
  int? shippingInfoId;
  int? status;
  String? statusName;
  String? statusNamee;
  DateTime? timeProcessed;
  DateTime? timePacked;
  DateTime? timeShipped;
  DateTime? timeDelivered;
  bool? onlineOrder;
  bool? delivery;
  double? shippingConst;
  int? noOfItems;
  double? total;
  ShippingInfo? shippingInfo;
  Customer? customer;
  Staff? staff;

  List<OrderDetail> orderDetails = [];

  OrderHeader();

  factory OrderHeader.fromJson(Map<String, dynamic> json) =>
      _$OrderHeaderFromJson(json);

  Map<String, dynamic> toJson() => _$OrderHeaderToJson(this);

  double getTotal() {
    double result = 0;

    orderDetails.forEach((element) => result += element.total());

    return result;
  }
}
